pandas/tests/series/test_cumulative.py

```diff
@@ -170,6 +170,53 @@ def test_cummethods_bool_in_object_dtype(self, method, expected):
         result = getattr(ser, method)()
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "method, order",
+        [
+            ["cummax", "abc"],
+            ["cummin", "cba"],
+        ],
+    )
+    def test_cummax_cummin_on_ordered_categorical(self, method, order):
+        # GH#52335
+        cat = pd.CategoricalDtype(list(order), ordered=True)
+        ser = pd.Series(
+            list("ababcab"), dtype=pd.CategoricalDtype(list(order), ordered=True)
+        )
+        result = getattr(ser, method)()
+        tm.assert_series_equal(result, pd.Series(list("abbbccc"), dtype=cat))
+
+    @pytest.mark.parametrize(
+        "method, order",
+        [
+            ["cummax", "abc"],
+            ["cummin", "cba"],
+        ],
+    )
+    def test_cummax_cummin_ordered_categorical_nan(self, method, order):
+        # GH#52335
+        ser = pd.Series(
+            ["a", np.nan, "b", "a", "c"],
+            dtype=pd.CategoricalDtype(list(order), ordered=True),
+        )
+        result = getattr(ser, method)(skipna=True)
+        tm.assert_series_equal(
+            result,
+            pd.Series(
+                ["a", np.nan, "b", "b", "c"],
+                dtype=pd.CategoricalDtype(list(order), ordered=True),
```

Comment:
Can you assign this to an `expected` variable?

Reply:
Assigned all expected series to `expected` variables